src/Worker/Grpc/GrpcDurableTaskWorker.Processor.cs

```diff
@@ -337,5 +345,69 @@ async Task OnRunActivityAsync(P.ActivityRequest request)
 
             await this.sidecar.CompleteActivityTaskAsync(response);
         }
+
+        async Task OnRunEntityBatchAsync(P.EntityBatchRequest request)
+        {
+            string instanceId = request.InstanceId;
+            var coreEntityId = DTCore.Entities.EntityId.FromString(request.InstanceId);
+            var entityId = new EntityId(coreEntityId.Name, coreEntityId.Key);
+
+            TaskContext innerContext = new(new OrchestrationInstance() { InstanceId = instanceId });
+            TaskName name = new(entityId.Name);
+
+            EntityBatchRequest batchRequest = request.ToEntityBatchRequest();
+            EntityBatchResult? batchResult;
+
+            try
+            {
+                await using AsyncServiceScope scope = this.worker.services.CreateAsyncScope();
+                if (this.worker.Factory.TryCreateEntity(name, scope.ServiceProvider, out ITaskEntity? entity))
+                {
+                    // Both the factory invocation and the RunAsync could involve user code and need to be handled as
+                    // part of try/catch.
+                    TaskEntity shim = this.shimFactory.CreateEntity(name, entity, entityId);
+                    batchResult = await shim.ExecuteOperationBatchAsync(batchRequest);
+                }
+                else
+                {
+                    // we could not find the entity. This is considered an application error,
+                    // so we return a non-retriable error-OperationResult for each operation in the batch.
+                    batchResult = new EntityBatchResult()
+                    {
+                        Actions = new List<OperationAction>(), // no actions
+                        EntityState = batchRequest.EntityState, // state is unmodified
+                        Results = Enumerable.Repeat(
+                            new OperationResult()
+                            {
+                                Result = null,
+                                ErrorMessage = "entity dispatch failed",
+                                FailureDetails = new FailureDetails(
+                                    errorType: "EntityTaskNotFound",
+                                    errorMessage: $"No entity task named '{name}' was found.",
+                                    stackTrace: null,
+                                    innerFailure: null,
+                                    isNonRetriable: true),
+                            },
+                            batchRequest.Operations!.Count).ToList(),
+                    };
+                }
+            }
+            catch (Exception frameworkException)
```

Comment:
This could use a log statement

Reply:
Good point. Logging is actually one of the missing things all across these PRs. I wasn't sure how much / what logging to add in general.

Reply:
We can go through logging later I suppose